components/brave_sync/brave_sync_service_impl.cc

```diff
@@ -484,7 +487,7 @@ void BraveSyncServiceImpl::OnResolvedPreferences(const RecordsList& records) {
         (record->deviceId == this_device_id &&
           record->action == jslib::SyncRecord::Action::A_DELETE &&
           actually_merged);
-      contains_only_one_device = sync_devices->size() < 2 &&
+      contains_only_one_device = sync_devices->have_less_than_two() &&
```

Comment:
I'm confused about this, if there's only one device and it has been deleted, then there are no devices, not one device